### ISSUE TITLE
[Handshake] Support an `will_next_write_payload_encrypted`

### ISSUE THREAD

**Issue by zserik** [mcginty/snow#84]
The `HandshakeState` should have some way to determine if the next payload will be encrypted, where false-negatives are acceptable (and will probably happen if `!is_my_turn()` and `read_message` wasn't called yet). e.g. A way to only send a payload with the next `write_message` if it will be encrypted.

**Comment by mcginty** [comment]
I agree that this definitely feels more useful than a "was_write_payload_encrypted", which was implemented because it's trivial.

To implement `will_write_payload_encrypt` feels hairier, because it basically involves doing a "dry run" of the `write_message` operation without changing internal state, which feels kind of dirty to me. That said, `snow` does now have checkpointing to recover from errors and avoid having an unknown internal state, and this feature *could* be used to implement this.

It's definitely going to be a more expensive function to call than its past-tense sibling though.
